ethsigner/core/src/main/java/tech/pegasys/ethsigner/core/signing/hashicorp/HashicorpSignerHelper.java

```diff
@@ -0,0 +1,138 @@
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.ethsigner.core.signing.hashicorp;
+
+import tech.pegasys.ethsigner.core.signing.CredentialTransactionSigner;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import io.vertx.core.Vertx;
+import io.vertx.core.http.HttpClient;
+import io.vertx.core.http.HttpClientRequest;
+import io.vertx.core.http.HttpMethod;
+import io.vertx.core.json.JsonObject;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.web3j.crypto.Credentials;
+
+public class HashicorpSignerHelper {
+
+  private static final Logger LOG = LogManager.getLogger();
+
+  private static final Vertx vertx = Vertx.vertx();
+  private static final HttpClient hashicorpVaultClient = vertx.createHttpClient();
+
+  private HashicorpSignerHelper() {}
+
+  public static CredentialTransactionSigner getSigner(HashicorpSignerConfig config) {
+    final String tokenString = readTokenFromFile(config);
+    if (tokenString == null) {
+      return null;
+    }
+    final String response = requestSecretFromVault(config, tokenString);
+    if (response == null) {
+      return null;
+    }
+    final Credentials credentials = extractCredentialsFromJson(response);
+    if (credentials == null) {
+      return null;
+    }
+    return new CredentialTransactionSigner(credentials);
+  }
+
+  private static String readTokenFromFile(HashicorpSignerConfig config) {
+    List<String> authFileLines;
+    try {
+      authFileLines = Files.readAllLines(config.getAuthFilePath());
+    } catch (IOException e) {
+      LOG.error(
+          "Unable to read file containing the authentication information for Hashicorp Vault.\n"
+              + config.getAuthFilePath()
+              + "\n"
+              + e.getMessage());
+      return null;
+    }
+    return authFileLines.get(0);
+  }
+
+  private static String requestSecretFromVault(HashicorpSignerConfig config, String tokenString) {
+    final String requestURI = "/v1" + config.getSigningKeyPath();
+
+    return getVaultResponse(config, tokenString, requestURI);
+  }
+
+  private static String getVaultResponse(
+      HashicorpSignerConfig config, String tokenString, String requestURI) {
+    final CompletableFuture<String> future = new CompletableFuture<>();
+    final HttpClientRequest request =
+        hashicorpVaultClient.request(
+            HttpMethod.GET,
+            config.getServerPort(),
+            config.getServerHost(),
+            requestURI,
+            rh -> {
+              rh.bodyHandler(
```

Comment:
could we reuse some of the VertxRequestTransmitter? Its not ideal - as the VRT expects there to be RoutingContext :/

Otherwise the lambda/failure could probably be broken out to separate functions rather than being inline.